src/System.Management.Automation/FormatAndOutput/common/FormatViewGenerator_Complex.cs

```diff
@@ -681,23 +683,46 @@ private void DisplayLeaf(object val, List<FormatValue> formatValueList)
         /// </summary>
         /// <param name="val">Object to verify.</param>
         /// <param name="level">Current level of recursion.</param>
-        /// <returns></returns>
-        private static bool TreatAsLeafNode(object val, TraversalInfo level)
+        /// <param name="scalarTypesToExpand">Array of names of types that should not be rendered as scalars.</param>
+        /// <returns>True if it has to be treated as leaf node.</returns>
+        private static bool TreatAsLeafNode(object val, TraversalInfo level, string[] scalarTypesToExpand)
         {
             if (level.Level >= level.MaxDepth || val == null)
+            {
                 return true;
+            }
 
-            return TreatAsScalarType(PSObject.GetTypeNames(val));
+            return TreatAsScalarType(PSObject.GetTypeNames(val), scalarTypesToExpand);
         }
 
         /// <summary>
         /// Treat as scalar check.
         /// </summary>
         /// <param name="typeNames">Name of the type to check.</param>
+        /// <param name="scalarTypesToExpand">Array of names of types that should not be rendered as scalars.</param>
         /// <returns>True if it has to be treated as a scalar.</returns>
-        private static bool TreatAsScalarType(Collection<string> typeNames)
+        private static bool TreatAsScalarType(Collection<string> typeNames, string[] scalarTypesToExpand)
+        {
+            return TypeIsScalarForComplexView(typeNames, scalarTypesToExpand) || DefaultScalarTypes.IsTypeInList(typeNames);
```

Comment:
1. I'd swap the condition since it makes no sense to change the behavior by user input.
2. And at first look it makes sense to put "Datetime" in DefaultScalarTypes (and enhance it with exclusion list). Perhaps it is useful for other code paths too.
3.  Perhaps it makes sense to use the s_defaultLeafTypes list in ExpandType parameter for IntelliSense.

Reply:
Regarding #2: It was not obvious to me that that behavior was desired. With Format-List, for example, a user may reasonably want to see all the properties of the type.